pkg/liqo-controller-manager/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -288,6 +285,7 @@ func (r *ForeignClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			klog.Error(err)
 			return ctrl.Result{}, err
 		}
+		r.ForeignClusters.Delete(foreignCluster.Status.TenantNamespace.Local)
```

Comment:
I am missing where the element is added to the map. Additionally, I would move the deletion to the check at the beginning of the reconcile, to catch all events (here it is called only with auto-discovery deletions).